lib/matplotlib/rcsetup.py

```diff
@@ -888,6 +872,36 @@ def validate_animation_writer_path(p):
         modules["matplotlib.animation"].writers.set_dirty()
     return p
 
+# A validator dedicated to the named line styles, based on the items in
+# ls_mapper, and a list of possible strings read from Line2D.set_linestyle
+validate_named_linestyle = ValidateInStrings('linestyle',
+                                             list(six.iterkeys(ls_mapper)) +
+                                             list(six.itervalues(ls_mapper)) +
+                                             ['None', 'none', ' ', ''],
+                                             ignorecase=False)
+
+# A validator for all possible line styles, the named ones *and*
+# the on-off ink sequences.
+def validate_linestyle(ls):
```

Comment:
can we make this a private function? I don't think it should be added to our public API.